test/util/k8sreporter/reporter.go

```diff
@@ -1,144 +1,53 @@
-
-func (r *KubernetesReporter) Report(sr types.SpecReport) {
-	r.Lock()
-	defer r.Unlock()
-	fmt.Fprintln(r.dumpOutput, "Starting dump for failed spec", sr.ContainerHierarchyTexts)
-	r.dump()
-	fmt.Fprintln(r.dumpOutput, "Finished dump for failed spec")
-}
-
-func (r *KubernetesReporter) dump() {
-	r.logNodes()
-	r.logPods("openshift-sriov-network-operator")
-	r.logPods(namespaces.Test)
-	r.logLogs(func(p *corev1.Pod) bool {
-		return !strings.HasPrefix(p.Name, "sriov-")
-	})
-	r.logSriovNodeState()
-	r.logNetworkPolicies()
-}
-
-func (r *KubernetesReporter) logPods(namespace string) {
-	fmt.Fprintf(r.dumpOutput, "Logging pods for %s", namespace)
-
-	pods, err := r.clients.Pods(namespace).List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
-		return
-	}
-
-	j, err := json.MarshalIndent(pods, "", "    ")
-	if err != nil {
-		fmt.Println("Failed to marshal pods", err)
-		return
-	}
-	fmt.Fprintln(r.dumpOutput, string(j))
-}
-
-func (r *KubernetesReporter) logNodes() {
-	fmt.Fprintf(r.dumpOutput, "Logging nodes")
-
-	nodes, err := r.clients.CoreV1Interface.Nodes().List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch nodes: %v\n", err)
-		return
-	}
-
-	j, err := json.MarshalIndent(nodes, "", "    ")
-	if err != nil {
-		fmt.Println("Failed to marshal nodes")
-		return
-	}
-	fmt.Fprintln(r.dumpOutput, string(j))
-}
-
-func (r *KubernetesReporter) logLogs(filterPods func(*corev1.Pod) bool) {
-	fmt.Fprintf(r.dumpOutput, "Logging pods logs")
-
-	pods, err := r.clients.Pods(corev1.NamespaceAll).List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
-		return
-	}
-
-	for _, pod := range pods.Items {
-		if filterPods(&pod) {
-			continue
-		}
-		for _, container := range pod.Spec.Containers {
-			logs, err := r.clients.Pods(pod.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{Container: container.Name}).DoRaw(context.Background())
-			if err == nil {
-				fmt.Fprintf(r.dumpOutput, "Dumping logs for pod %s-%s-%s", pod.Namespace, pod.Name, container.Name)
-				fmt.Fprintln(r.dumpOutput, string(logs))
-			}
+func New(reportPath string) (*kniK8sReporter.KubernetesReporter, error) {
+	addToScheme := func(s *runtime.Scheme) error {
+		err := sriovv1.AddToScheme(s)
+		if err != nil {
+			return err
 		}
+		return nil
 	}
-}
-
-func (r *KubernetesReporter) logNetworkPolicies() {
-	fmt.Fprintf(r.dumpOutput, "Logging network policies")
-
-	policies := sriovv1.SriovNetworkNodePolicyList{}
-	err := r.clients.List(context.Background(),
-		&policies,
-		runtimeclient.InNamespace("openshift-sriov-network-operator"))
 
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch network policies: %v\n", err)
-		return
+	dumpNamespace := func(ns string) bool {
+		switch {
+		case ns == namespaces.Test:
+			return true
+		case ns == "openshift-sriov-network-operator":
```

Comment:
ohhh I miss this one in the u/s review??
there is a variable for the operator namespace please use that one and not this string.
you will need to open the PR u/s first

Reply:
@SchSeba I am looking at the code and don't see a variable that is available for use.
I mean, there is a private const in the controllers package, and a private var in the conformance/tests package.
Do you propose to export the one in the conformance/tests package?

Reply:
I  think @SchSeba is referring to the `OPEATOR_NAMESPACE` environment variable:
It's used in other places in the test suite:

- [test/util/clean/clean.go#L18](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/08badf64b9781f772e38ac2eb29315501e93e0b0/test/util/clean/clean.go#L18)
- [test/conformance/tests/init.go#L15](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/08badf64b9781f772e38ac2eb29315501e93e0b0/test/conformance/tests/init.go#L15)
- [test/validation/tests/test_validation.go#L28](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/08badf64b9781f772e38ac2eb29315501e93e0b0/test/validation/tests/test_validation.go#L28)

Reply:
Thank you. I filed this: https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/506

Reply:
The PR upstream merged, I rebased this one.